homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -83,7 +84,7 @@ async def async_step_user(
         if user_input is not None:
             unique_id = f"{user_input[CONF_REGION]}-{user_input[CONF_USERNAME]}"
 
-            if self.source != SOURCE_REAUTH:
+            if self.source not in {SOURCE_REAUTH, SOURCE_RECONFIGURE}:
```

Comment:
What happens if the user changes the region or the username inside the reauth/reconfigure flow?

It seems to me that the entry data would no longer match the unique_id...

Here is an example where I think the unique_id helper suggested in #127565 would be useful

```python
    await self.async_set_unique_id(unique_id)
    if self.source in {SOURCE_REAUTH, SOURCE_RECONFIGURE}:
        self._abort_if_unique_id_configured()
    else:
        self._abort_if_unique_id_mismatch()
```

Reply:
Yes, this makes absolute sense. As the helper is not available right now, I got inspired by this PR and implemented it directly in the config flow. If this changes at some point, I'm happy to adjust.